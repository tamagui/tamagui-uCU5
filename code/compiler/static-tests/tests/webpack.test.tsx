import '@expo/match-media'

import { render } from '@testing-library/react'
import TestRenderer from 'react-test-renderer'
import { describe, expect, test } from 'vitest'

/**
 * disabled for now but we really need to bring this back
 */

function getTest(name: string) {
  const app = require('./spec/out/out-webpack')
  const App = app[name]
  if (!App) {
    throw new Error(`No app found: ${name} in ${Object.keys(app || {})}`)
  }
  const Provider = app.Provider
  return {
    get Element() {
      return (props) => (
        <Provider>
          <App {...props} />
        </Provider>
      )
    },
    get renderer() {
      return TestRenderer.create(
        <Provider>
          <App conditional={true} />
        </Provider>
      )
    },
    get rendererFalse() {
      return TestRenderer.create(
        <Provider>
          <App conditional={false} />
        </Provider>
      )
    },
  }
}

describe('webpack-tests', () => {
  test('1. extracts to a div for simple views, flat transforms', () => {
    const test1 = getTest('Test1')
    const out = test1.renderer.toJSON()!
    expect(out).toMatchSnapshot()
  })

  test('2. extracts className for complex views but keeps other props', () => {
    const test2 = getTest('Test2')
    const out = test2.renderer.toJSON()!
    expect(out).toMatchSnapshot()
    const outFalse = test2.rendererFalse.toJSON()
    expect(outFalse).toMatchSnapshot()
  })

  // test('3. places className correctly given a single spread', async () => {
  //   const { Element } = getTest('Test3')
  //   const out = render(<Element />)
  //   const list = [...out.container.firstChild?.['classList']]
  //   expect(list).toMatchSnapshot()
  // })

  // test('4. leaves dynamic variables', async () => {
  //   const { renderer, Element } = getTest('Test4')
  //   const out = render(<Element />)
  //   const firstChild = out.container.firstChild!
  //   const classList = [...firstChild['classList']]
  //   expect(classList).toMatchSnapshot()
  //   const r = renderer.toJSON()
  //   expect(r).toMatchSnapshot()
  // })

  test('5. spread conditional', async () => {
    const test5 = getTest('Test5')
    const out = test5.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('6. spread ternary', async () => {
    const test6 = getTest('Test6')
    expect(test6.renderer.toJSON()).toMatchSnapshot()
    expect(test6.rendererFalse.toJSON()).toMatchSnapshot()
  })

  test('7. ternary + data-is', async () => {
    const test7 = getTest('Test7')
    const out = test7.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('8. styleExpansions', async () => {
    const test8 = getTest('Test8')
    const out = test8.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('9. combines with classname', async () => {
    const test9 = getTest('Test9')
    const out = test9.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('10. extracts Text', async () => {
    const test10 = getTest('Test10')
    const out = test10.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  // test('11. combines everything', async () => {
  //   const { Element } = getTest('Test11')
  //   const out = render(<Element conditional={false} />)
  //   const firstChild = out.container.firstChild!
  //   const classList = [...firstChild['classList']]
  //   expect(classList).toMatchSnapshot()
  // })

  test('12. te  rnary multiple on same key', async () => {
    const test12 = getTest('Test12')
    expect(test12.renderer.toJSON()).toMatchSnapshot()
  })

  // test('13. text with complex conditional and local vars', async () => {
  // const test13 } = getTest('{')
  //   // console.log('test13', test13.renderer!.toTree())
  //   t.is(1, 1)
  // })

  test('14. extracts pseudo styles and evaluates constants', async () => {
    const test14 = getTest('Test14')
    const out = test14.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('15. extracts spacer (complex expansion)', () => {
    const test15 = getTest('Test15')
    const out = test15.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('16. deopt when spreading multiple', () => {
    const test16 = getTest('Test16')
    const out = test16.renderer.toJSON()
    expect(out).toMatchSnapshot()
  })

  test('17. variant default false flattens properly', () => {
    const out = getTest('TestVariantDefaultFalseOn').renderer.toJSON()
    const outCn = out.children[0].children[0].props.className

    const out2 = getTest('TestVariantDefaultFalseOff').renderer.toJSON()
    const out2Cn = out2.children[0].children[0].props.className

    expect(outCn).to.not.contain(`_pt-1316330145`)
    expect(out2Cn).to.contain(`_pt-1316330145`)
  })
})
