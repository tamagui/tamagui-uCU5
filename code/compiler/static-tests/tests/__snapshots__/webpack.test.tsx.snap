// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`webpack-tests > 1. extracts to a div for simple views, flat transforms 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="  _dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-1 _fg-1 _btlr-100px _btrr-100px _bbrr-100px _bblr-100px _bg-red _bxsh-469092149 _scale-0hover-2  test1"
    >
      <span
        className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 font_body "
      >
        hello world
      </span>
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 2. extracts className for complex views but keeps other props 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="  _dsp-flex _ai-stretch _fd-row _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _ox-hidden _oy-hidden  who"
      onClick={[Function]}
    />
    <div
      className="_dsp-flex _ai-stretch _fd-row _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _pt-2px _pb-2px _t-0px ease-in-out-top _t--14px _bg-fff35 "
    >
      <span
        className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 font_body "
      >
        hello world
      </span>
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 2. extracts className for complex views but keeps other props 2`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="  _dsp-flex _ai-stretch _fd-row _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _ox-hidden _oy-hidden  who"
      onClick={[Function]}
    />
    <div
      className="_dsp-flex _ai-stretch _fd-row _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _bg-00035 _pt-2px _pb-2px _t-0px ease-in-out-top "
    >
      <span
        className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 font_body "
      >
        hello world
      </span>
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 5. spread conditional 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _ox-hidden _oy-hidden _bg-blue "
    >
      <div
        className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _ox-hidden _oy-hidden hello-world _bg-blue "
      />
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 6. spread ternary 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _ox-hidden _oy-hidden _bg-blue "
    >
      <span
        className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 font_body "
      >
        hello world
      </span>
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 6. spread ternary 2`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _ox-hidden _oy-hidden _bg-red "
    >
      <span
        className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 font_body "
      >
        hello world
      </span>
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 7. ternary + data-is 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _fs-0 _pr-11px _pl-11px _pb-10px _w-6637 _miw-500px _maw-3037"
      style={{}}
    >
      <div
        className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _w-10px _h-10px "
      />
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 8. styleExpansions 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="  _dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _mih-0px _miw-0px _fs-0 _pos-relative _t-0px _l-0px _r-0px _b-0px "
    >
      <div
        className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _bg-red"
        style={{}}
      >
        <div
          className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _mih-0px _miw-0px _fs-0 _pos-relative _t-0px _l-0px _r-0px _b-0px _bg-white _col-blue"
          style={{}}
        />
      </div>
    </div>
  </span>
</span>
`;

exports[`webpack-tests > 9. combines with classname 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="  _dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _pt-15px _pb-15px  home-top-dish"
    />
  </span>
</span>
`;

exports[`webpack-tests > 10. extracts Text 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <span
      className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 _fos-10px font_body "
    >
      <span
        className="  _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-675002279 _ussel-auto font_body "
      >
        <span
          className="font_body _ff-299667014 _dsp-inline _bxs-border-box _ww-break-word _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _col-00035 _fos-15px _fow-600"
          style={{}}
        >
          hello
        </span>
      </span>
    </span>
  </span>
</span>
`;

exports[`webpack-tests > 12. te  rnary multiple on same key 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0 _o-1 _transform- "
    />
  </span>
</span>
`;

exports[`webpack-tests > 14. extracts pseudo styles and evaluates constants 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_bg-0hover-red _bg-0active-blue _dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-0"
      style={{}}
    />
  </span>
</span>
`;

exports[`webpack-tests > 15. extracts spacer (complex expansion) 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <span
      className="  is_Spacer _dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-1316330176 _miw-1316330176 _fs-0 _w-1316330176 _h-1316330176 _pe-none "
    />
    <span
      className="  is_Spacer _dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-10px _miw-10px _fs-1 _pe-none _fg-1 _w-10px _h-10px "
    />
  </span>
</span>
`;

exports[`webpack-tests > 16. deopt when spreading multiple 1`] = `
<span
  className=" _dsp_contents"
>
  <span
    className=" t_dark _dsp_contents is_Theme"
    style={
      {
        "color": "var(--color)",
      }
    }
  >
    <div
      className="_dsp-flex _ai-stretch _fd-column _fb-auto _bxs-border-box _pos-relative _mih-0px _miw-0px _fs-1 _fg-1 _bg-red"
      conditional={true}
      style={{}}
    />
  </span>
</span>
`;
